Firebase/Core/FIRBundleUtil.m

```diff
@@ -45,9 +45,10 @@ + (NSArray *)relevantURLSchemes {
   return result;
 }
 
-+ (BOOL)hasBundleIdentifier:(NSString *)bundleIdentifier inBundles:(NSArray *)bundles {
++ (BOOL)hasBundleIdentifierPrefix:(NSString *)bundleIdentifier inBundles:(NSArray *)bundles {
   for (NSBundle *bundle in bundles) {
-    if ([bundle.bundleIdentifier isEqualToString:bundleIdentifier]) {
+    // This allows app extensions that have the app's bundle as their prefix to pass this test.
+    if ([bundleIdentifier hasPrefix:bundle.bundleIdentifier]) {
```

Comment:
We want to be a bit more careful here I think - this would match on things like `com.firebase.foobar` or `com.firebase.foo.staging` where it should only match on `com.firebase.foo`.

We could change the method call to `matchesBundleIdentifier:inBundles:` which could do an exact string match if it's not an extension, but if it is, compare to ensure it's the entire bundleID + a period + an extension identifier.

There's an existing `isAppExtension` call in `GoogleUtilities` I believe we could use, and if that's true, separate by `.`, drop the last one, and re-assemble to verify that it's identical.